### DISCUSSION TITLE
Feature: Allow disabling XA enforcement introduced with v26

### DISCUSSION THREAD

**Discussion by BerGer23** [keycloak/keycloak#34879] [Ideas]
Keycloak 26 introduced the enforcement of XA whenever more than one datasource is used through the new BuildStep [checkMultipleDatasourcesUseXA](https://github.com/keycloak/keycloak/blame/637ca2e13848c74444d98604918670217247bf64/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java#L306).

While this seems like a good idea generally, it prevents certain scenarios from working that do not _really_ require distributed transactions nor have the option to enable them.
In the case of the setup I am looking to upgrade to Keycloak v26, the secondary database is essentially a read-only user storage provider.

Since the AWS JDBC Wrapper has to be used here, enabling XA is not an option - the scenario is described in more detail at https://github.com/keycloak/keycloak/discussions/34837.

It would be great if Keycloak exposed a mechanism to disable the XA enforcement (that should likely come with a "use at your own risk" type of description in the documentation)

**Comment by shawkins** [comment]
Let's just refer to #34837 in a new issue, rather than starting a new discussion.

**Comment by BerGer23** [comment]
Got it.
